build/lib/accounting/write_jobs.py

```diff
@@ -0,0 +1,135 @@
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+from __future__ import print_function
+
+import sys
+import sqlite3
+import pandas as pd
+import time
+
+import flux.job
+import flux.constants
+import flux.util
+from flux.core.inner import raw
+
+
+def write_to_db(inactive):
+    # open connection to database
+    print("Opening JobCompletion DB...")
+    conn = sqlite3.connect("JobCompletion.db")
+    print("Opened JobCompletion DB successfully\n")
+
+    # insert values into DB one at a time by getting the values from
+    # each job completion tuple
+    print("Inserting values into DB...\n")
+    for job in inactive:
+        conn.execute(
+            """
+            INSERT OR IGNORE INTO inactive
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);
+            """,
+            (
+                job["t_submit"],
+                job["name"],
+                job["t_run"],
+                job["t_cleanup"],
+                job["userid"],
+                job["ntasks"],
+                job["t_inactive"],
+                job["t_depend"],
+                job["priority"],
+                job["state"],
+                job["t_sched"],
+                job["id"],
+            ),
+        )
+        print("insertion into inactive table successful")
+
+    # commit changes
+    conn.commit()
+
+    # make sure the writes to DB were successful and we can
+    # retrieve them
+    sel_statement = "SELECT * FROM inactive"
+    print(pd.read_sql_query(sel_statement, conn))
+    print()
+
+    conn.close()
+
+
+def save_last_job():
+    # open connection to database
+    print("Opening JobCompletion DB...")
+    conn = sqlite3.connect("JobCompletion.db")
+    print("Opened JobCompletion DB successfully\n")
+
+    # get the last submitted job and query its t_submit and job id
+    print("Getting last submitted job...")
+    sel_statement = "SELECT id, t_submit FROM inactive ORDER BY t_inactive DESC LIMIT 1"
+    cursor = conn.cursor()
+    cursor.execute(sel_statement)
+    records = cursor.fetchall()
+
+    for row in records:
+        last_job_id = row[0]
+        last_t_inactive = row[1]
+
+    print("Last job id:", last_job_id, "t_inactive:", last_t_inactive)
+    print("Query complete\n")
+
+    # write the last job's job id to a temporary text file for retrieval later
+    # if the file doesn't exist yet, it will be created
+    print("writing last job id to tmp file...")
+    f = open("tmp_id.txt", "w")
+    f.truncate(0)
+    f.write(last_job_id + "\n")
+    f.close()
+    print("write complete\n")
+
+    # <demo> show contents of last job's job id
+    print("showing contents of tmp file...")
+    f = open("tmp_id.txt", "r")
+    for line in f:
+        print(line)
+
+
+def main():
```

Comment:
I recommend decorating this main function with our `util.CLIMain` decorator, which captures uncaught exceptions and pretty-prints them (i.e., avoids dumping a massive stack-trace unless you set a specific environment variable).

Here is an example of that in `flux-mini.py`: https://github.com/flux-framework/flux-core/blob/a06041b4a3e981a11e5b46c219beb7853e00a214/src/cmd/flux-mini.py#L297